doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1100 @@
+178    PSU_FAULT                        Wed Feb 10 08:08:24 2021    /psu/2 experienced a fault. 
+
+sonic# show alarm recent 5min 
+Sq No  Name                             Timestamp                   Description
+-----  ------------------------------   --------------------------- --------------
+2291   TEMPERATURE_EXCEEDED             Wed Feb 10 18:08:24 2021    temperatureCrossedThreshold: Current temperature of sensor/2 is 76 degrees
+
+sonic# show alarm sequence-id from 2290 to 2300
+Sq No  Name                             Timestamp                   Description
+-----  ------------------------------   --------------------------- --------------
+2291   TEMPERATURE_EXCEEDED             Wed Feb 10 18:08:24 2021    temperatureCrossedThreshold: Current temperature of sensor/2 is 76 degrees
+
+```
+
+### 3.3.3 REST API Support
+
+sonic REST links:
+*  /restconf/data/sonic-event:sonic-event/EVENT/EVENT_LIST
+*  /restconf/data/sonic-event:sonic-event/EVENT_STATS/EVENT_STATS_LIST
+*  /restconf/data/sonic-alarm:sonic-alarm/ALARM/ALARM_LIST
+*  /restconf/data/sonic-alarm:sonic-alarm/ALARM_STATS/ALARM_STATS_LIST
+
+openconfig REST links:
+*  /restconf/data/openconfig-system:system/openconfig-events:events
+*  /restconf/data/openconfig-system:system/openconfig-events:event-stats
+*  /restconf/data/openconfig-system:system/alarms
+*  /restconf/data/openconfig-system:system/openconfig-alarms-ext:alarm-stats
+
+# 4 Flow Diagrams
+![Sequence Diagram](event-alarm-framework-seqdiag.png)
+
+# 5 Warm Boot Support
+## 5.1 Application warm boot
+Applications confirming to the warm boot, should have stored their state and compare current values against previous values.
+Such compliant application also "remembers" that it raised an event before for a specific condition.
+They would 
+*  not raise alarms/events for the same condition that it raised pre warm boot
+*  clear those alarms once current state of a particular condition is recovered (by comparing against the stored state). 
+
+## 5.2 eventd warm boot
+Records from applications are stored in a table, called EVENTPUBSUB.
+By using table of type ProducerTable, records that are being written will be queued when the consumer (eventd) is down.
+
+During normal operation, eventd reads, processes whenever a new record is added to the table.
+
+When eventd is restarted, events and alarms raised by applications will be waiting in a queue while eventd is coming up.
+When eventd eventually comes back up, it reads those records in the queue.
+
+# 6 Scalability
+In this feature, scalability applies to Event History Table (EVENT). As it is persistent and it records every event generated on the system, to protect
+against it growing indefinitely, user can limit its size through a manifest file.
+By default, the size of Event History Table is set to 40k events or events for 30 days - after which, older records are discarded to make way for new records.
+
```

Comment:
Please add a reference to the show techsupport. I believe we need the configuration as well as the 'show' commands in it.

Reply:
show tech will include output of 'show event recent 1hr' and 'show alarm all' commands. I will update the HLD.

Reply:
Updated.